From b641401558cca6a8f35fbceebb52a5110f8d5d8f Mon Sep 17 00:00:00 2001
From: Masaharu Hayakawa <masaharu.hayakawa.ry@renesas.com>
Date: Thu, 21 Sep 2017 16:38:26 +0900
Subject: [PATCH 199/909] mmc: tmio: Add SDCLK enable after reset

commit af91a0de8919774c92949a432e0158ef95f37707 from
git://git.kernel.org/pub/scm/linux/kernel/git/horms/renesas-bsp.git

The driver performs a software reset in error handling
when timeout error occurred.
SDCLK was disabled by software reset, so this patch enables
SDCLK after software reset.

Signed-off-by: Masaharu Hayakawa <masaharu.hayakawa.ry@renesas.com>
Signed-off-by: Takeshi Saito <takeshi.saito.xv@renesas.com>
Signed-off-by: Meng Li <Meng.Li@windriver.com>
---
 drivers/mmc/host/tmio_mmc_core.c | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/drivers/mmc/host/tmio_mmc_core.c b/drivers/mmc/host/tmio_mmc_core.c
index 82a4b24ee1c9..f52fef359906 100644
--- a/drivers/mmc/host/tmio_mmc_core.c
+++ b/drivers/mmc/host/tmio_mmc_core.c
@@ -250,6 +250,7 @@ static void tmio_mmc_reset_work(struct work_struct *work)
 						  delayed_reset_work.work);
 	struct mmc_request *mrq;
 	unsigned long flags;
+	u16 clk;
 
 	spin_lock_irqsave(&host->lock, flags);
 	mrq = host->mrq;
@@ -282,7 +283,9 @@ static void tmio_mmc_reset_work(struct work_struct *work)
 
 	spin_unlock_irqrestore(&host->lock, flags);
 
+	clk = sd_ctrl_read16(host, CTL_SD_CARD_CLK_CTL) & CLK_CTL_DIV_MASK;
 	tmio_mmc_reset(host);
+	sd_ctrl_write16(host, CTL_SD_CARD_CLK_CTL, clk | CLK_CTL_SCLKEN);
 
 	/* Ready for new calls */
 	host->mrq = NULL;
-- 
2.17.1

